rust/src/storage/azure.rs

```diff
@@ -89,75 +79,73 @@ impl Default for ADLSGen2Backend {
     }
 }
 
+#[async_trait::async_trait]
 impl StorageBackend for ADLSGen2Backend {
-    fn head_obj(&self, path: &str) -> Result<ObjectMeta, StorageError> {
+    async fn head_obj(&self, path: &str) -> Result<ObjectMeta, StorageError> {
         let obj = parse_uri(path)?.into_adlsgen2_object()?;
+        let path = path.to_string();
         debug!("Getting properties for {}", path);
-        let properties = Self::gen_tokio_rt().block_on(
-            self.client
-                .get_blob_properties()
-                .with_container_name(obj.file_system)
-                .with_blob_name(obj.path)
-                .finalize(),
-        )?;
-        Ok(ObjectMeta {
-            path: path.to_string(),
-            modified: properties
-                .blob
-                .last_modified
-                .expect("Last-Modified should never be None for committed blobs"),
-        })
+        let properties = self
+            .client
+            .get_blob_properties()
+            .with_container_name(obj.file_system)
+            .with_blob_name(obj.path)
+            .finalize()
+            .await
+            .unwrap();
```

Comment:
AzureError from trait has already been implemented for StorageError, I think we should be able to replace the unwrap here with a`?`?

Reply:
Good spot! Fixed.